sharding/scheme.go

```diff
@@ -29,36 +29,50 @@ import (
 var (
 	// ErrToLessThanFrom returned when to is less than from
 	ErrToLessThanFrom = errors.New("to is less than from")
+	// ErrNoShards returned when shard set is empty
+	ErrNoShards = errors.New("empty shard set")
+	// ErrDuplicateShards returned when shard set is empty
+	ErrDuplicateShards = errors.New("duplicate shards")
 )
 
 type shardScheme struct {
-	from uint32
-	to   uint32
-	fn   m3db.HashFn
+	shards []uint32
+	fn     m3db.HashFn
 }
 
-// NewShardScheme creates a new sharding scheme, from and to are inclusive
-func NewShardScheme(from, to uint32, fn m3db.HashFn) (m3db.ShardScheme, error) {
+// NewShardSchemeWithRange creates a new sharding scheme with a range of consecutive shards, from and to are inclusive
+func NewShardSchemeWithRange(from, to uint32, fn m3db.HashFn) (m3db.ShardScheme, error) {
```

Comment:
No I mean call it `NewShardSchemeFromRange` instead of `NewShardSchemeWithRange`